src/units/abilities.cpp

```diff
@@ -1557,6 +1557,11 @@ bool attack_type::special_active_impl(const_attack_ptr self_attack, const_attack
 	temporary_facing other_facing(other, other_loc.get_relative_dir(self_loc));
 
 	// Filter poison, plague, drain, first strike
+	bool bool_lambda = ((whom == AFFECT_SELF) || ((whom == AFFECT_EITHER) && special_affects_self(special, is_attacker))) ? true : false;
+	unit_const_ptr lambda_unit = bool_lambda ? other : self;
```

Comment:
`lambda_unit` is unused.

For the other `lambda_*` variables, I'm trying to think of suggestions for better names. For 1.17, I'd suggest `victim` "For status conditions, the unit that will be harmed by this condition".

Reply:
changes done